cli/npm/managed/resolution.rs

```diff
@@ -223,34 +223,42 @@ impl NpmResolution {
   /// Resolves a package requirement for deno graph. This should only be
   /// called by deno_graph's NpmResolver or for resolving packages in
   /// a package.json
-  pub fn resolve_pkg_req_as_pending(
+  pub async fn resolve_pkg_req_as_pending_with_info(
     &self,
     pkg_req: &PackageReq,
+    pkg_info: &NpmPackageInfo,
   ) -> Result<PackageNv, NpmPackageVersionResolutionError> {
-    // we should always have this because it should have been cached before here
-    let package_info = self.api.get_cached_package_info(&pkg_req.name).unwrap();
-    self.resolve_pkg_req_as_pending_with_info(pkg_req, &package_info)
-  }
-
-  /// Resolves a package requirement for deno graph. This should only be
-  /// called by deno_graph's NpmResolver or for resolving packages in
-  /// a package.json
-  pub fn resolve_pkg_req_as_pending_with_info(
-    &self,
-    pkg_req: &PackageReq,
-    package_info: &NpmPackageInfo,
-  ) -> Result<PackageNv, NpmPackageVersionResolutionError> {
-    debug_assert_eq!(pkg_req.name, package_info.name);
+    debug_assert_eq!(pkg_req.name, pkg_info.name);
+    let _permit = self.update_queue.acquire().await;
```

Comment:
Bug fix here. Previously two threads could be modifying the snapshot at the same time.

Reply:
I'm just creating an `SyncReadAsyncWriteLock` to make this mistake impossible in the future (will land in a future pr).